libs/platform/src/lib/components/list/base-list-item.ts

```diff
@@ -0,0 +1,370 @@
+import {
+    ElementRef, HostBinding, Input, ChangeDetectorRef, EventEmitter,
+    Output, HostListener, ViewChild, AfterViewChecked, OnInit, Directive, TemplateRef
+} from '@angular/core';
+
+import { ENTER, SPACE } from '@angular/cdk/keycodes';
+import { CheckboxComponent, RadioButtonComponent } from '@fundamental-ngx/core';
+import { ContentDensity } from '../form/form-control';
+import { BaseComponent } from '../base';
+import { ListConfig } from './list.config';
+
+let nextListItemId = 0;
+export type StatusType = 'negative' | 'critical' | 'positive' | 'informative';
+export type SelectionType = 'none' | 'multi' | 'single' | 'delete';
+export type ListType = 'inactive' | 'active' | 'detail';
+
+/** Base interface for a list variant definition.
+ *  Captures a list item template definition. */
+export interface ItemDef {
+    templateRef: TemplateRef<any>;
+}
+
+@Directive({ selector: '[fdpItemDef]' })
+export class ListItemDef implements ItemDef {
+    constructor(/** @docs-private */ public templateRef: TemplateRef<any>) { }
+}
+
+/**
+ * Interface for defining more actions on list item
+ * secondary
+ */
+interface SecondaryActionItem {
+    icon: string;
+    isButton: boolean;
+}
+/**
+ * This class contains common properties used across list Item components.
+ * this can be extended to reduce the code duplication across list Item components.
+ */
+@Directive()
+export class BaseListItem extends BaseComponent implements OnInit, AfterViewChecked {
+
+    /** sets the list item in bold text
+     * which respresents unread data */
+    @Input()
+    unRead: boolean;
+
+    /** To invert the status of secondary text*/
+    @Input()
+    inverted = false;
+
+    /** Whether there is no data inside list item */
+    @Input()
+    noDataText?: string;
```

Comment:
Do you need to mek this properties optional with question mark?

Reply:
updated